cnudie/util.py

```diff
@@ -42,6 +42,13 @@ def determine_main_source_for_component(
     return component.sources[0]
 
 
+@dataclasses.dataclass
+class LabelDiff:
+    labels_only_left: typing.List[cm.Label] = dataclasses.field(default_factory=list)
+    labels_only_right: typing.List[cm.Label] = dataclasses.field(default_factory=list)
+    label_pairs_changed: typing.List[typing.Tuple[cm.Label, cm.Label]] = dataclasses.field(default_factory=list) # noqa:E501
```

Comment:
stylistic nit: you might define a (reusable) module-variable e.g. `empty_list`, thus also achieving shorter lines